src/Orleans/Runtime/SiloStatus.cs

```diff
@@ -41,5 +41,13 @@ public static bool IsTerminating(this SiloStatus siloStatus)
         {
             return siloStatus == SiloStatus.ShuttingDown || siloStatus == SiloStatus.Stopping || siloStatus == SiloStatus.Dead;
         }
+
+        /// <summary>
+        /// Return true if silo is terminating or missing.
+        /// </summary>
+        public static bool IsUnavailable(this SiloStatus siloStatus)
+        {
+            return siloStatus.IsTerminating() || siloStatus == SiloStatus.None;
```

Comment:
In a general sense, I think that's ok.
In this specific case it should be.
This check is only made if we can't talk to the other silo.  If it is there but just not in our membership service yet, then we should be able to talk to it.